### DISCUSSION TITLE
Does PnP Core SDK Manage Throttling for Custom API Calls?

### DISCUSSION THREAD

**Discussion by bajce** [pnp/pnpcore#1456] [Q&A] [answered]
I'm using the PnP Core SDK to make custom API calls as explained here: [Custom API Requests with PnP Core SDK](https://pnp.github.io/pnpcore/using-the-sdk/basics-customapirequests.html). I want to know if the SDK still manages throttling (like retries and waiting times) when we use custom APIs. Do I need to do anything extra for this?

Thanks for your help!

**Comment by jansenbe** [comment, marked as answer]
@bajce : any request made using the HttpClient instances used by PnP Core SDK does handle throttling as we plugin the throttling logic at that level. So, yes, custom API calls are also retried in case of throttling

**Comment by bajce** [comment]
@jansenbe That's great. Thank you for your answer!
